keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,488 @@
+
+## Motivation
+
+For end-users who would like to run unprivileged containers using user namespaces _nested inside_ CRI containers, we need an option to have a `ProcMount`.
+That is, we need an option to designate explicitly turn off masking and setting read-only of paths so that we can mount `/proc` in the nested container as an unprivileged user.
+
+Please see the following filed issues for more information:
+- [opencontainers/runc#1658](https://github.com/opencontainers/runc/issues/1658#issuecomment-373122073)
+- [moby/moby#36597](https://github.com/moby/moby/issues/36597)
+- [moby/moby#36644](https://github.com/moby/moby/pull/36644)
+
+### Goals
+
+- Add the ProcMountType option to the pod API, allowing users to override the proc mount behavior in a container.
+
+### Non-Goals
+
+## Proposal
+
+
+Add a new `string` type field named `ProcMountType` will hold the viable
+options for `procMount` to the `SecurityContext`
+definition.
+
+By default,`procMount` is `default`, aka the same behavior as today and the
+paths are masked.
+
+This will look like the following in the spec:
+
+```go
+type ProcMountType string
+
+const (
+    // DefaultProcMount uses the container runtime default ProcType.  Most 
+    // container runtimes mask certain paths in /proc to avoid accidental security
+    // exposure of special devices or information.
+    DefaultProcMount ProcMountType = "Default"
+
+    // UnmaskedProcMount bypasses the default masking behavior of the container
+    // runtime and ensures the newly created /proc the container stays in tact with
+    // no modifications.  
+    UnmaskedProcMount ProcMountType = "Unmasked"
+)
+
+procMount *ProcMountType
+```
+
+This requires changes to the CRI runtime integrations so that kubelet will add the specific `unmasked` option.
+
+Note: Pod Security Admission will also need to be taught how to handle this option.
+[Luckily, this has been done after the feature was introduced.](https://github.com/kubernetes/kubernetes/pull/103340)
+
+Unmasking the paths in `/proc` option really only makes sense for when a user is nesting unprivileged containers with user namespaces as it will allow more information
+than is necessary to the program running in the container spawned by kubernetes.
+
+It should be noted that this is different that the host /proc. It is still a newly mounted /proc just the container runtimes will not mask the paths.
+
+Since the only use case for this option is to run unprivileged nested containers, this option should only be allowed or used if the user in the container is not `root`.
+This can be easily enforced with `MustRunAs`.
+
+Since the user inside is still unprivileged, doing things to `/proc` would be off limits regardless, since linux user support already prevents this.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a cluster admin, I would like a way to nest containers within containers and get access to proc entries as if the top level container was running
```

Comment:
AFAICS you are not interested in getting access to those masked procfs entries.
The actual reason you want to unmask them is because the kernel doesn't allow mounting a new procfs without unmasking them, no?